### ISSUE TITLE
KuCoin Futures: fetch_order_status returns 'canceled' status while order still exists on exchange

### ISSUE THREAD

**Issue by RoscoeTheDog** [ccxt/ccxt#14575]
This somewhat hacky block of code places a take profit order, waits a second, then checks the order again while blocking  until it has been confirmed OK on the exchange. I can see the `self.place_futures_order` returns a dict with a `status` of 'None'. When I fetch the same order using `fetch_order_status`, it returns with a status of `'canceled'` but I can see the take-profit order existing on the exchange still. I verified the order id being passed is the same as the returned from the `create_order` call when fetching it from the exchange. My intuition is leading me to believe that since the order exists on the exchange previous to the `place_futures_order` call, I can't place additional take-profits orders on my position because it would be greater than my position size. If I am long position, I can place multiple buy/long orders on the exchange and it is no problem since that's incrementing relative position size. If I decrement relative position size, I seem to have issues here. Again this is just a running theory. I'm hoping someone can clarify this behavior here.

Since a take profit exists on the exchange for the full position size, then would any sell orders get rejected with a 'canceled' status from the exchange? I think this behavior should be better documented somewhere. Seeing a random cancelation status when `create_order` returns a valid dictionary structure is a bit misleading behavior. It makes the end-user to believe it had registered on the exchange successfully, especially if it includes other info like price, volume etc but not status. It also creates performance bottlenecks if users have to verify an order every time after creating one since the 'status' field always returns `None` upon creating orders with Kucoin Futures.

Again-- I don't know if this is the actually the required case yet. Did I find a bug or is this behavior intentional?

```
                        while order_confirmation is None:
                            if not response:
                                response = self.place_futures_order(active_config['symbol'], 'limit', direction, Decimal(str(self.record.TARGET_VOLUME)), Decimal(str(self.record.TARGET_PRICE)), params={'closeOrder': True}) # https://stackoverflow.com/questions/71529052/python-cctx-kucoin-futures-take-profit

                            # wait for the rate limit so we don't endlessly get caught here flooding messages to the exchange
                            time.sleep(self.rate_limit / 1000)

                            if response and isinstance(response, dict):
                                if response.get('id'):
                                    self.record.TAKE_PROFIT_RESPONSE = response
                                    # confirm the order on the exchange to prevent it flooding endless orders. placing it endlessly will never let the exchange place/register the order officially.
                                    try:
                                        if not order_confirmation:
                                            order_confirmation = self.exchange.fetch_order(self.record.TAKE_PROFIT_RESPONSE.get('id'), active_config['symbol'])
                                            time.sleep(self.rate_limit / 1000)
                                            # TODO: Find why Kucoin is flagging newly created orders as canceled
                                            if order_confirmation.get('status') == 'canceled':
                                                order_confirmation['status'] = 'open'
                                                self.logger.warning('take profit order was flagged "canceled" by exchange incorrectly. Verify limit order on exchange!')
                                    except Exception as e:
                                        order_confirmation = None
                                        self.logger.error('error confirming take-profit order', confirmation=order_confirmation, exc_info=e)
                                        time.sleep(self.rate_limit / 1000)
```
                                        
                                        
                                      
                                      
```
    def place_futures_order(self, symbol, type, side, amount, price, params={}):
        response = None
        # https://github.com/ccxt/ccxt/issues/13356
        record = self.get_request_limiter_time()
        now = time.time()
        if (now - float(record.LAST_CHECK)) > self.rate_limit/10000:

            try:
                response = self.exchange.create_order(symbol, type, side, float(amount), float(price), params)
                self.update_request_limiter_time()
            except Exception as e:
                self.update_request_limiter_time()
                self.logger.error('could not create order', response=response)
                return None

            # TODO: make request checks more explicit and return the response
            info = response.get('info')
            if int(info.get('code')) == 200000:     # https://docs.kucoin.com/futures/#requests
                response['price'] = str(Decimal(str(price)))
                response['average'] = str(response['price'])
                response['amount'] = str(Decimal(str(amount)))

                markets = self.exchange.load_markets(symbol)
                target_market = markets[symbol]
                contract_size = target_market['contractSize']
                precision = target_market['precision']
                prec_amount = precision['amount']

                if side == 'buy':
                    response['cost'] = Decimal(str(response['average'])) * Decimal(str(amount)) * Decimal(str(contract_size)) * Decimal(str((1 + self.taker_fee)))
                if side == 'sell':
                    response['cost'] = (Decimal(str(response['average'])) * Decimal(str(amount)) * Decimal(str(contract_size)) * Decimal(str((1 + self.taker_fee)))) / Decimal(str(amount))

                response['cost'] = str(response['cost'])  # cast to string to serialize to JSON on tables
                response['filled'] = response.get('amount')
                if str.lower(type) == 'market':
                    response['status'] = 'closed'
            elif int(info.get('code')) == 300009:
                response['status'] = 'closed'
                return response
            else:
                self.logger.error('the order could not be placed', code=info.get('code'))
                return None

        return response
```

**Comment by samgermain** [comment]
HI @RoscoeTheDog

There's sort of a lot to go through in this question. To answer your question more quickly, would you be able to provide a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

Thank you

**Comment by RoscoeTheDog** [comment]
I am still unsure what caused this issue when I first encountered it, but I posted this back in early august and lost the context to the problem. My code is trading as expected right now, and I don't think I'll have the time to hunt down or remember what caused this.

I'm going to close this issue for now seeing as I don't have time to investigate it further and my software seems to be running as expected now after lots more dev time. I'm assuming it was my developer-error and not with the ccxt module.
